src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ViewDeclarationOrder/ViewDeclarationOrderAnalyzer.cs

```diff
@@ -1,77 +1,124 @@
 ﻿using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
-using Acuminator.Utilities.Roslyn;
-using Acuminator.Utilities.Roslyn.Semantic;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Acuminator.Utilities.Common;
+using Acuminator.Utilities.Roslyn;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Acuminator.Analyzers.StaticAnalysis.PXGraph;
+using Acuminator.Utilities.Roslyn.Semantic.PXGraph;
+using Acuminator.Utilities;
 
 namespace Acuminator.Analyzers.StaticAnalysis.ViewDeclarationOrder
 {
-    [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public class ViewDeclarationOrderAnalyzer : PXDiagnosticAnalyzer
-    {
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
-            ImmutableArray.Create(Descriptors.PX1004_ViewDeclarationOrder, Descriptors.PX1006_ViewDeclarationOrder);
+	/// <summary>
+	/// An analyzer for the order of view declaration in graph/graph extension.
+	/// This diagnostic works only for simple class hierarchy where the depth of the inheritance is equal to 2: object -> DAC -> DerivedDAC.
+	/// </summary>
+	public class ViewDeclarationOrderAnalyzer : IPXGraphAnalyzer
+	{
+		private const string InitCacheMappingMethodName = "InitCacheMapping";
```

Comment:
It is better to use nameof() instead of string constant

Reply:
This was done with the consideration that we will move from PX.Data to string constants...  somewhere in the future. In that case nameof won't work.

Reply:
As far as I know, currently there is no plans to do it.